geometry/spline.cpp

```diff
@@ -168,6 +169,7 @@ bool Spline::iterator::IsAttached() const
 
 void Spline::iterator::Advance(double step)
 {
+  m_length += std::abs(step);
```

Comment:
In general this is not the same as the previous GetLength() implementation when we are moving backward.
The semantics of the GetLength() function is GetLengthFromBegin().

Since the GetLength() function is used only in gps track, I'd prefer to remove it from here and accumulate it in GpsTrackRenderer::RenderTrack as a separate var.

Reply:
Or alternatively, you can:
- rename GetLength -> GetLengthFromBegin
- put m_length += step here
- put ASSERT(m_length >=0, (m_length)) after

Reply:
Thanks! I moved the length accumulation to the RenderTrack and removed the GetLength() from the iterator at all.